### PR TITLE
Guaranteed synchronous reactor order execution

### PR DESCRIPTION
**Pull request by darrunategui**
#### Why 
Reactors were stored in a `Set` upon registration. Although ruby `Set`s preserve order, this is not guaranteed in the future and depends on the internal implementation of a `Set`. If we want to preserve order we should be using an array.

Preserving order lets us guarantee the order or execution of synchronous reactors (the order in which they were registered).

#### What changed 
* Use an array to store reactors
* Update shared examples to accept either a single reactor or multiple reactors

### CODE REVIEW

**Review by ferenc-nagy — APPROVED** [review]

File: lib/eventsimple/event_dispatcher.rb [hunk 1]
```diff
@@ -77,16 +78,18 @@ class ReactorSet
...
       def add_sync(reactors)
-        @sync += reactors
+        new_reactors = reactors - @sync
+        @sync += new_reactors
       end
 
       def add_async(reactors)
-        @async += reactors
+        new_reactors = reactors - @async
+        @async += new_reactors
```
Comment (desheikh) [inline lib/eventsimple/event_dispatcher.rb:92]: I think this is equivalent to using the [| operator](https://ruby-doc.org/core-2.5.0/Array.html#method-i-7C).

```suggestion
        @async |= reactors
```

Comment (darrunategui) [reply lib/eventsimple/event_dispatcher.rb:92]: oh nice! I didn't know this operator existed. TY

File: spec/dummy/spec/components/user_component/events/created_spec.rb [hunk 2]
```diff
@@ -21,6 +21,12 @@
     it_behaves_like 'an event which synchronously dispatches',
       UserComponent::Reactors::Created::SyncReactor
 
+    it_behaves_like 'an event which synchronously dispatches in order',
```
[the remark this answers is not in the document]
Comment (darrunategui) [reply spec/dummy/spec/components/user_component/events/created_spec.rb]: Good idea. Fixed in latest commit

**Review by desheikh — APPROVED** [review]
Thanks David 🙏

**Pull request merged**